services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3523,17 +3522,26 @@ private boolean hasFatalDataValidationError(PushMonitor pushMonitor, String topi
     }
   }
 
-  private void safeDeleteRTTopic(String clusterName, String storeName, Store store) {
+  private void safeDeleteRTTopic(String clusterName, String storeName) {
+    Store store = getHelixVeniceClusterResources(clusterName).getStoreMetadataRepository().getStore(storeName);
+    if (canDeleteRTTopic(clusterName, storeName, store)) {
+      deleteRTTopic(clusterName, storeName);
+    } else {
+      LOGGER.warn("Topic deletion for topic: {} is delayed.", Version.composeRealTimeTopic(storeName));
+    }
+  }
+
+  public boolean canDeleteRTTopic(String clusterName, String storeName, Store store) {
     // Perform RT cleanup checks for batch only store that used to be hybrid. Check the local versions first
     // to see if any version is still using RT and then also check other fabrics before deleting the RT. Since
     // we perform this check everytime when a store version is deleted we can afford to do best effort
     // approach if some fabrics are unavailable or out of sync (temporarily).
-    boolean canDeleteRT = !Version.containsHybridVersion(store.getVersions());
+    return !Version.containsHybridVersion(store.getVersions()) && canDeleteRTTopic(clusterName, storeName);
+  }
+
+  public boolean canDeleteRTTopic(String clusterName, String storeName) {
```

Comment:
Can we rename the function somehow to reflect the fact that  we are querying the child controllers for their opinions?